src/main/java/com/google/devtools/build/lib/rules/cpp/ShowIncludesFilter.java

```diff
@@ -148,10 +148,19 @@ public static class FilterShowIncludesOutputStream extends FilterOutputStream {
                 StandardCharsets.UTF_8) // Spanish
             );
     private final String sourceFileName;
+    private boolean sawPotentialUnsupportedShowIncludesLine;
     // Grab everything under the execroot base so that external repository header files are covered
     // in the sibling repository layout.
     private static final Pattern EXECROOT_BASE_HEADER_PATTERN =
         Pattern.compile(".*execroot\\\\(?<headerPath>.*)");
+    // Match a line of the form "fooo: bar:   C:\some\path\file.h". As this is relatively generic,
+    // we require the line to include an absolute path with drive letter. If remote workers rewrite
+    // the path to a relative one, we won't match it, but it is unlikely that such setups use an
+    // unsupported encoding. We also exclude any matches that contain numbers: MSVC warnings and
+    // errors always contain numbers, but the /showIncludes output doesn't in any encoding since all
+    // codepages are ASCII-compatible.
+    private static final Pattern POTENTIAL_UNSUPPORTED_SHOW_INCLUDES_LINE =
+        Pattern.compile("[^:0-9]+:\\s+[^:0-9]+:\\s+[A-Za-z]:\\\\.*");
```

Comment:
Hmm, I wondering if this this pattern to general that it'll give false positives. For example
`C:\path\to\file.cpp(10): error C2146: syntax error: missing ';' before identifier 'x'` seems to match?

Reply:
It doesn't match, for two reasons: We check that something path-like comes after the second `:` and we also exclude any numbers before the first two colons. This is still hacky, but any kind of error or warning message I could find had numbers present in these places.

Reply:
I see, do we need to match `execroot`?

Reply:
That is an interesting question: As far as I understand we do report non-execroot dependencies, so we would also want to catch `/showIncludes` output parsing issues for those. But missing them wouldn't affect incremental correctness in the same way as missing an execroot header would, so we could require an `execroot` path to lower the false positive rate. I just don't know whether it would actually do that since most paths should be execroot paths anyway.

Reply:
I added a check for `execroot` as well as no `:` in paths (forbidden on Windows). I also added tests for the "near matches" and fixed the condition under which we report the error.

Reply:
Nice!